src/sentry/workflow_engine/tasks.py

```diff
@@ -140,22 +182,121 @@ def process_workflows_event(
     has_escalated: bool,
     **kwargs,
 ) -> None:
-    event = fetch_event(event_id, project_id)
-    if event is None:
-        logger.error("Event not found", extra={"event_id": event_id, "project_id": project_id})
-        return
+    from sentry.workflow_engine.processors.workflow import process_workflows
 
-    occurrence = IssueOccurrence.fetch(occurrence_id, project_id) if occurrence_id else None
-    group = Group.objects.get(id=group_id)
-    group_event = GroupEvent.from_event(event, group)
-    group_event.occurrence = occurrence
-    event_data = WorkflowEventData(
+    event_data = build_workflow_event_data(
+        project_id=project_id,
+        event_id=event_id,
+        group_id=group_id,
+        occurrence_id=occurrence_id,
+        group_state=group_state,
         has_reappeared=has_reappeared,
         has_escalated=has_escalated,
-        group_state=group_state,
-        event=group_event,
-        group=group,
     )
     process_workflows(event_data)
 
     metrics.incr("workflow_engine.tasks.process_workflow_task_executed", sample_rate=1.0)
+
+
+def build_trigger_action_task_params(action, detector, event_data: WorkflowEventData):
+    """Build parameters for trigger_action.delay() call."""
+    event_id = None
+    activity_id = None
+    occurrence_id = None
+
+    if isinstance(event_data.event, GroupEvent):
+        event_id = event_data.event.event_id
+        occurrence_id = event_data.event.occurrence_id
+    elif isinstance(event_data.event, Activity):
+        activity_id = event_data.event.id
+
+    return {
+        "action_id": action.id,
+        "detector_id": detector.id,
+        "workflow_id": getattr(action, "workflow_id", None),
+        "event_id": event_id,
+        "activity_id": activity_id,
+        "group_id": event_data.event.group_id,
+        "occurrence_id": occurrence_id,
+        "group_state": event_data.group_state,
+        "has_reappeared": event_data.has_reappeared,
+        "has_escalated": event_data.has_escalated,
+        "workflow_env_id": event_data.workflow_env.id if event_data.workflow_env else None,
+    }
```

Comment:
is all of this data required to rebuild the context? for example, do we need to pass the `workflow_id` here since we're getting it from the action?